torch/CMakeLists.txt

```diff
@@ -274,6 +274,7 @@ if (USE_NCCL)
     list(APPEND TORCH_PYTHON_SRCS
       ${TORCH_SRC_DIR}/csrc/cuda/python_nccl.cpp)
     list(APPEND TORCH_PYTHON_COMPILE_DEFINITIONS USE_NCCL)
+    list(APPEND TORCH_PYTHON_LINK_LIBRARIES __caffe2_nccl)
```

Comment:
@pritamdamania87 I saw you move this to `caffe2/CMakeLists.txt` in #27124. The new location of this link was later removed by #30315. Do you see any concern of adding it back here?

also cc @ezyang

Reply:
Don't recall the exact reason for this, but it might have been to fix a build issue. I think this move should be fine as long as CI is happy.